grakn-engine/src/main/java/ai/grakn/engine/GraknEngineServer.java

```diff
@@ -49,125 +49,132 @@
 
 import static ai.grakn.engine.util.ConfigProperties.TASK_MANAGER_IMPLEMENTATION;
 import static org.apache.commons.lang.exception.ExceptionUtils.getFullStackTrace;
-import static spark.Spark.awaitInitialization;
-import static spark.Spark.before;
-import static spark.Spark.exception;
-import static spark.Spark.halt;
-import static spark.Spark.ipAddress;
-import static spark.Spark.port;
-import static spark.Spark.staticFiles;
-import static spark.Spark.webSocket;
-import static spark.Spark.webSocketIdleTimeoutMillis;
 
 /**
  * Main class in charge to start a web server and all the REST controllers.
  *
  * @author Marco Scoppetta
  */
 
-public class GraknEngineServer {
+public class GraknEngineServer implements AutoCloseable {
     private static final ConfigProperties prop = ConfigProperties.getInstance();
 
-    private static final EngineID ENGINE_ID = EngineID.me();
     private static final Logger LOG = LoggerFactory.getLogger(GraknEngineServer.class);
     private static final int WEBSOCKET_TIMEOUT = 3600000;
     private static final Set<String> unauthenticatedEndPoints = new HashSet<>(Arrays.asList(
             REST.WebPath.NEW_SESSION_URI,
             REST.WebPath.REMOTE_SHELL_URI,
             REST.WebPath.GRAPH_FACTORY_URI,
             REST.WebPath.IS_PASSWORD_PROTECTED_URI));
-
     public static final boolean isPasswordProtected = prop.getPropertyAsBool(ConfigProperties.PASSWORD_PROTECTED_PROPERTY);
-    private static TaskManager taskManager;
+
+    private final EngineID engineId = EngineID.me();
+    private final int port;
+    private final Service spark = Service.ignite();
+    private final TaskManager taskManager;
+
+    private GraknEngineServer(String taskManagerClass, int port) {
+        taskManager = startTaskManager(taskManagerClass);
+        this.port = port;
+        startHTTP();
+        startPostprocessing();
+        printStartMessage(prop.getProperty(ConfigProperties.SERVER_HOST_NAME), prop.getProperty(ConfigProperties.SERVER_PORT_NUMBER), prop.getLogFilePath());
+    }
 
     public static void main(String[] args) {
+        mainWithServer();
+    }
+
+    public static GraknEngineServer mainWithServer() {
+        // Start Engine
+        int port = prop.getPropertyAsInt(ConfigProperties.SERVER_PORT_NUMBER);
+        String taskManagerClass = prop.getProperty(TASK_MANAGER_IMPLEMENTATION);
+        GraknEngineServer server = start(taskManagerClass, port);
+
         // close GraknEngineServer on SIGTERM
-        Thread closeThread = new Thread(GraknEngineServer::stop, "GraknEngineServer-shutdown");
+        Thread closeThread = new Thread(server::close, "GraknEngineServer-shutdown");
```

Comment:
Should we be adding the shutdown hook here? In our tests we will be adding the hook many times, which may not be ideal.